via/resources.py

```diff
@@ -0,0 +1,27 @@
+"""Context objects for views."""
+from pyramid.httpexceptions import HTTPBadRequest
+
+# pylint: disable=too-few-public-methods
+
+
+class URLRoute:
+    """Methods for routes which accept a 'url'."""
+
+    def __init__(self, request):
+        self._request = request
+
+    @property
```

Comment:
I think it's a little weird for `@property`'s to raise exceptions. See PEP 8:

> Avoid using properties for computationally expensive operations; the attribute notation makes the caller believe that access is (relatively) cheap.

Of course that's not talking about raising exceptions but I think the same reasoning applies: the attribute notation makes it look, from the caller's point of view, like a simple attribute access and so it's surprising that it would raise.

It's a PITA to make a mock property access raise an exception with the `mock` library, if you wanted to add tests for the views like "it does X if context.url raises HTTPBadRequest" (in this PR you didn't write those tests for the views, and I don't think they're necessary in this case either).

When writing the unit tests for a property that raises you'll end up falsely triggering PyLint's `pointless-statement` warning, as you did in your test above, which wouldn't happen if it were a method call.

I think the fact that both mock and PyLint don't expect property accesses to raise and make it awkward to test ones that do, is telling us something.

Reply:
I've changed this over to a method.